Sprint 04

MWD650 opened the pull request:
Было сложно, но я старался ^^

MWD650 committed:
Add files via upload

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,188 @@ import UIKit
+    private func showNextQuestionOrResults() {
+      if currentQuestionIndex == questions.count - 1 { // - 1 потому что индекс
+        //начинается с 0, а длинна массива — с 1 показать результат квиза
+       //  print("конец") // проверка
```
лучше удалять такой закоменченный код

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,188 @@ import UIKit
+         let firstQuestion = questions[currentQuestionIndex]
+         let viewModel = convert(model: firstQuestion)
+              
+              show(quiz: viewModel)
```
едет форматирование. Предлагаю использовать `cmd+A` + `ctrl+I`

MWD650 replied:
ok

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,188 @@ import UIKit
+           imageView.layer.borderWidth = 8
+           imageView.layer.borderColor = isCorrect ? UIColor.green.cgColor : UIColor.red.cgColor
+     
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1.0) {  self.showNextQuestionOrResults()
```
лучше переносить начало кложуры в следующую строку

MWD650 replied:
ok

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,188 @@ import UIKit
+    @IBAction private func yesButtonClicked(_ sender: UIButton) {
+        let currentQuestion = questions[currentQuestionIndex]
+        let givenAnswer = true
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
```
61-62 можно сократить до 
`showAnswerResult(isCorrect: currentQuestion.correctAnswer)`

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,188 @@ import UIKit
+        let currentQuestion = questions[currentQuestionIndex]
+        let givenAnswer = false
+        showAnswerResult(isCorrect: givenAnswer == currentQuestion.correctAnswer)
+        sender.isEnabled = false
```
круто, что сразу обработали. Можно вынести это в начало функции `showAnswerResult(`. Так будет меньше повторяющегося кода.

mmbabaev commented on MovieQuiz/Presentation/Base.lproj/Main.storyboard:
```diff
@@ -4,9 +4,18 @@
     <dependencies>
```
Не хватает bottom constraint к safe area у корневой stack view (корневой)

mmbabaev commented on MovieQuiz/Presentation/Base.lproj/Main.storyboard:
```diff
@@ -4,9 +4,18 @@
     <dependencies>
         <deployment identifier="iOS"/>
```
нет округления картинки

mmbabaev commented on MovieQuiz/Presentation/Base.lproj/Main.storyboard:
```diff
@@ -4,9 +4,18 @@
     <dependencies>
         <deployment identifier="iOS"/>
         <plugIn identifier="com.apple.InterfaceBuilder.IBCocoaTouchPlugin" version="20020"/>
+        <capability name="Named colors" minToolsVersion="9.0"/>
```
после начала игры остается обводка последнего ответа

mmbabaev commented on MovieQuiz/Presentation/MovieQuizViewController.swift:
```diff
@@ -2,71 +2,188 @@ import UIKit
+            }
+           imageView.layer.masksToBounds = true
+           imageView.layer.borderWidth = 8
+           imageView.layer.borderColor = isCorrect ? UIColor.green.cgColor : UIColor.red.cgColor
```
цвета не соответствуют макетам

mmbabaev commented on MovieQuiz/Presentation/Base.lproj/Main.storyboard:
```diff
@@ -15,13 +24,122 @@
+        <namedColor name="YP Black">
+            <color red="0.10199999809265137" green="0.10599999874830246" blue="0.13300000131130219" alpha="1" colorSpace="custom" customColorSpace="sRGB"/>
+        </namedColor>
+    </resources>
```
давайте сделаем у картинки другой `contentMode = .aspectFill`

mmbabaev commented on MovieQuiz/AppDelegate.swift:
```diff
@@ -11,6 +11,7 @@ import UIKit
 final class AppDelegate: UIResponder, UIApplicationDelegate {
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch.
+        //sleep(1)
```
можно удалить

MWD650 replied:
ok

mmbabaev commented on MovieQuiz/Resources/Assets.xcassets/Images/Defolt.imageset/Contents.json:
```diff
@@ -0,0 +1,22 @@
+{
```
нужна ли эта картинка?

mmbabaev commented on MovieQuiz/Resources/Assets.xcassets/Images/Small.imageset/Contents.json:
```diff
@@ -0,0 +1,21 @@
+  "info" : {
+    "author" : "xcode",
+    "version" : 1
+  }
```
картинку кнопки тоже можно удалить

MWD650 committed:
Add files via upload

MWD650 committed:
Delete MovieQuiz/Resources/MovieQuizFonts directory

MWD650 committed:
Merge pull request #3 from MWD650/sprint_04_fix

Внес правки

MWD650 committed:
Merge pull request #4 from MWD650/sprint_04_fix

Sprint 04 fix

MWD650 committed:
Так и не разобрался как удалить PR, вроде ее можно только закрыть

mmbabaev commented:
Также шрифты не меняются в приложении. В учебнике допущена ошибка, поэтому этот пункт необязателен. 
На скрине показал как нужно сделать: 
1) Шрифты добавлены в Group (серая папка)
2) Шрифтам проставлена галочка Target Membership  (справа)

<img width="1356" alt="Screen Shot 2022-11-17 at 20 15 01" src="https://user-images.githubusercontent.com/12600807/202513128-d74c5681-7a76-4778-ae5b-0fbcd44f0893.png">

Полезная статья, которая помогла мне разобраться в свое время: 
https://codewithchris.com/common-mistakes-with-adding-custom-fonts-to-your-ios-app/

MWD650 committed:
Подзакруглил

MWD650 committed:
Так, дозакруглил округление , и добавил шрифты как в той статье, спасибо большое за наводку

MWD650 committed:
Опять у меня все перепуталось и поломалось(

MWD650 committed:
Папка с шрифтами, комиться

mmbabaev commented:
Александр, круто, что разобрались в итоге со шрифтами. Округление есть, работу принял. 

Заметил еще, что цвет зеленый ставится неправильный, нужно исправить обязательно. 
Проблема в том, что вы добавили цвет как "YB Green " (с пробелом в конце)

mmbabaev commented:
еще цвет должен быть black из добавленных цветов

MWD650 committed:
Поправил цвета, (должен быть black из добавленных цветов, Заметил еще, что цвет зеленый ставится неправильный, нужно исправить обязательно.
Проблема в том, что вы добавили цвет как "YB Green " (с пробелом в конце))
Поправил закругление в первом вопросе (его не было)
